codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieEncoder.java

```diff
@@ -163,17 +191,32 @@ public String encode(Cookie cookie) {
      * @return the corresponding bunch of Set-Cookie headers
      */
     public List<String> encode(Iterable<? extends Cookie> cookies) {
-        if (!checkNotNull(cookies, "cookies").iterator().hasNext()) {
+        Iterator<? extends Cookie> cookiesIt = checkNotNull(cookies, "cookies").iterator();
+        if (!cookiesIt.hasNext()) {
             return Collections.emptyList();
         }
 
-        List<String> encoded = new ArrayList<String>();
-        for (Cookie c : cookies) {
-            if (c == null) {
-                break;
+        if (strict) {
+            Map<String, Cookie> nameToCookie = new HashMap<>();
+            while (cookiesIt.hasNext()) {
+                Cookie c = cookiesIt.next();
+                if (c == null) {
+                    break;
+                }
+                nameToCookie.put(c.name(), encode(c));
+            }            
+            return new ArrayList<>(nameToCookie.values());
```

Comment:
Mmm... The method returns a List there, which somehow implies that it keeps the Iterable input order. That's no longer the case with your change.

I wonder if returning a List is really useful, in which case your PR needs to be fixed, or if it doesn't, in which case we should change the method signature to return a Collection, and get rid of the extra allocation and directly return the Map values.